Remove unnecessary directory checks in delta lake

pajaks opened the pull request:
<!-- Thank you for submitting a pull request! Find more information in our development guide at https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md and contact us on #dev in Slack. -->
<!-- Provide an overview of the PR for maintainers and reviewers. -->
## Description
Origin of this change is to migrate to `TrinoFileSystem`
Relates to https://github.com/trinodb/trino/issues/16020


<!-- Provide details that would help an engineer who is unfamiliar with this part of the code. -->
## Additional context and related issues



<!-- Mark the appropriate option with an (x). Propose a release note if you can. -->
## Release notes

(x ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

ebyhr commented:
/test-with-secrets sha=6866240318db76360cebff85c9ca0432f9251395

pajaks commented on plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java:
```diff
@@ -710,7 +701,6 @@ public void createTable(ConnectorSession session, ConnectorTableMetadata tableMe
             external = false;
         }
         Path targetPath = new Path(location);
-        ensurePathExists(session, targetPath);
```
This check can be skipped because directory is created implicitly when first file is created.

findepi replied:
This should be part of the commit message

pajaks commented on plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java:
```diff
@@ -961,7 +926,7 @@ public Optional<ConnectorOutputMetadata> finishCreateTable(
                 .map(dataFileInfoCodec::fromJson)
                 .collect(toImmutableList());
 
-        Table table = buildTable(session, schemaTableName(schemaName, tableName), location, getExternalPath(new HdfsContext(session), location), handle.isExternal());
```
This seems to be redundant check, as individual files (and implicitly directories) are created at this point. If location would be file it should fail during directory creation earlier.

findepi commented on plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java:
```diff
@@ -808,29 +798,13 @@ private static Map<String, String> deltaTableProperties(ConnectorSession session
     }
 
-    private static void setDeltaStorageFormat(Table.Builder tableBuilder, String location, Path targetPath)
+    private static void setDeltaStorageFormat(Table.Builder tableBuilder, String location)
```
Can you make it as a separate (eg first) commit?

ebyhr commented:
nit: I would include "in Delta Lake" or something to commit messages.

pajaks commented:
Commit message changed.

pajaks commented:
@ebyhr @findepi Could you run tests with secretes?

ebyhr commented:
@pajaks Was there any change since last review except for the commit title?

pajaks commented:
> @pajaks Was there any change since last review except for the commit title? We don't need to run CI if the change is only commit title.

Only commit title changes since last review, but there was other changes since last run with secret.

ebyhr commented:
/test-with-secrets sha=ddf985373d9fcabbe8299798b453526ef69de737

findepi commented:
> The CI workflow run with tests that require additional secrets finished as failure: https://github.com/trinodb/trino/actions/runs/4342318738

```
Error: ] Could not find the selected project in the reactor: :trino-ignite @ 
Error:  Could not find the selected project in the reactor: :trino-ignite -> [Help 1]
Error:  
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error:  
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MavenExecutionException
Error: Process completed with exit code 1.
```

this means that `test-other-modules` didn't work at all. 
but these don't use secrets, so proceeding to merge
